test/unit_tests/autoqasm/test_api.py

```diff
@@ -1266,3 +1268,27 @@ def test(int[32] a, int[32] b) {
 test(2, 3);
 test(4, 5);"""
     assert main.build().to_ir() == expected
+
+
+def test_subroutine_call_with_reserved_keyword():
+    """Test that subroutine call works with reserved keyword as a variable name"""
+
+    @aq.subroutine
+    def make_input_state(input: int, theta: float):
```

Comment:
Yes, this is a good point, but probably a rare case. It would be worth opening an issue to track this - how do we handle the case where the user manually names variables both `xyz` and `xyz_`, where `xyz` is some reserved word.

Reply:
after the keyword gets sanitized, we can check if the name is present already in args or kwargs or `inspect.signature(func_name).parameters.keys()` or we could use `locals()` to check if the new name collides with other variable name. If we need to keep the user defined name then we need to think of some other way of changing the reserved keyword name

Reply:
Is there a way to add comments in the code before/after generating?

Reply:
I opened #30 for this issue.